pkg/satellite/stream/stream.go

```diff
@@ -178,6 +191,17 @@ func (ss *satelliteStream) recvLoop() {
 	b := backoff.NewExponentialBackOff()
 	b.MaxElapsedTime = MaxElapsedTime
 
+	// Initialize auto close timer
+	receivingBytes := false
+	var timestampLastByteReceived *timestamp.Timestamp
+	autoCloseTimer := time.AfterFunc(ss.autoCloseDelay, func() {
+		// Timer reached the end of the delay period. Closing stream and exiting
+		ss.Close()
+		panic("Stream auto-close conditions met - exiting")
```

Comment:
Basically, `panic()` should be used for critical bug or unexpected error. It would be better to exit gracefully if possible.

Reply:
+1 on 'exit gracefully'